tests/Unit/PdfTest.php

```diff
@@ -0,0 +1,262 @@
+    }
+
+    protected function uploadPdf(string $uploadToken): TestResponse
+    {
+        return $this->json('POST', route('v1.upload', [$uploadToken]), [
+            'file' => UploadedFile::fake()->createWithContent(self::PDF_NAME, File::get(base_path('tests/data/test.pdf'))),
+            'identifier' => self::IDENTIFIER
+        ]);
+    }
+
+    #[Test]
+    #[Depends('ensurePdfUploadSlotCanBeReserved')]
+    public function ensurePdfCanBeUploaded(string $uploadToken)
+    {
+        $uploadResponse = $this->uploadPdf($uploadToken);
+
+        $uploadResponse->assertCreated();
+
+        $media = Media::whereIdentifier(self::IDENTIFIER)->first();
+        $version = $media->Versions()->whereNumber($uploadResponse['version'])->first();
+
+        Storage::disk(config('transmorpher.disks.originals'))->assertExists($version->originalFilePath());
+
+        return $version;
+    }
+
+    #[Test]
+    #[Depends('ensurePdfUploadSlotCanBeReserved')]
+    #[Depends('ensurePdfCanBeUploaded')]
+    public function ensureUploadTokenIsInvalidatedAfterUpload(string $uploadToken)
+    {
+        $this->uploadPdf($uploadToken)->assertNotFound();
+    }
+
+    protected function getOriginal(Version $version): TestResponse
+    {
+        return $this->get(route('v1.getDocumentOriginal', [$version->Media, $version]));
+    }
+
+    protected function getDerivative(Version $version, ?string $transformations = null): TestResponse
+    {
+        return $this->get(route('getDocumentDerivative', [$this->user->name, $version->Media, $transformations]));
+    }
+
+    #[Test]
+    #[Depends('ensurePdfCanBeUploaded')]
+    public function ensurePdfOriginalCanBeDownloaded(Version $version)
+    {
+        $response = $this->getOriginal($version);
+
+        $response->assertOk();
+        $response->assertHeader('Content-Type', 'application/pdf');
+
+        return $version;
+    }
+
+    /**
+     * This test is not part of the data provider below, so it can be depended on in other tests.
+     * It covers the case where no transformations are applied.
+     */
+    #[Test]
+    #[Depends('ensurePdfOriginalCanBeDownloaded')]
+    public function ensurePdfDerivativeCanBeDownloaded(Version $version)
+    {
+        $response = $this->getDerivative($version);
+
+        $response->assertStatus(200);
+        $response->assertHeader('Content-Type', 'application/pdf');
+
+        return $version;
+    }
+
+    #[Test]
+    #[Depends('ensurePdfDerivativeCanBeDownloaded')]
+    #[DataProvider('providePdfTransformationStrings')]
+    public function ensurePdfDerivativeImagesCanBeDownloaded(string $transformations, int $expectedStatusCode, callable|string $expectedContentType, Version $version)
+    {
+        $response = $this->getDerivative($version, $transformations);
+
+        $expectedContentType = (is_callable($expectedContentType) ? $expectedContentType() : $expectedContentType) === 'image/jpg' ? 'image/jpeg' : $expectedContentType;
```

Comment:
please either decodegolf this into meaningful variablenames or add some documentation about what this is supposed to do.